mypy/types.py

```diff
@@ -539,7 +551,7 @@ def with_name(self, name: str) -> 'FunctionLike': pass
 
     @classmethod
     def deserialize(cls, data: JsonDict) -> 'FunctionLike':
-        return cast(FunctionLike, super().deserialize(data))
+        return cast(FunctionLike, deserialize_type(data))
```

Comment:
Hm... I think this method should be deleted and inlined in the one call site (in FuncDef.deserialize()).